Library/Homebrew/dev-cmd/bottle.rb

```diff
@@ -331,9 +331,8 @@ def bottle_formula(f, args:)
     else
       ohai "Determining #{f.full_name} bottle rebuild..."
       versions = FormulaVersions.new(f)
-      rebuilds = versions.bottle_version_map("origin/master")[f.pkg_version]
-      rebuilds.pop if rebuilds.last.to_i.positive?
-      rebuilds.empty? ? 0 : rebuilds.max.to_i + 1
+      rebuilds = versions.most_recent_rebuild_for_pkg_version("origin/master", f.pkg_version)
```

Comment:
I think this could be simpler still and avoid using `FormulaVersions` at all; `f.bottle_specification.rebuild` is the existing rebuild and you can do similar comparison with `origin/HEAD` of a Git repository to figure out the most recent rebuild/version (so look at a single command rather than traversing).

Reply:
Yep, that makes sense! Maybe if the pkg_version is less than origin/head, give a warning that they are rolling back the version and if that's what they really intend they are on their own for the rebuild number.